VotingApplication/VotingApplication.Web/Scripts/Controllers/Poll/Results/ResultsController.js

```diff
@@ -0,0 +1,72 @@
+﻿(function () {
+    'use strict';
+
+    angular
+        .module('VoteOn-Results')
+        .controller('ResultsController', ResultsController);
+
+    ResultsController.$inject = ['$scope', '$routeParams', '$interval', 'ResultsService'];
+
+    function ResultsController($scope, $routeParams, $interval, ResultsService) {
+
+        $scope.pollId = $routeParams['pollId'];
+        $scope.resultsBarChart = null;
+        $scope.results = [];
+        $scope.winners = [];
+        $scope.loaded = false;
+
+        var updateTimer = false;
+
+        activate();
+
+        function activate() {
+
+            updateResults();
+
+            updateTimer = $interval(updateResults, 5000);
+            $scope.$on('$destroy', clearTimer);
+        }
+
+        function clearTimer() {
+            $interval.cancel(updateTimer);
+        }
+
+        function updateResults() {
+            ResultsService.getResults($scope.pollId)
+           .then(function (data) {
+               if (data) {
+                   $scope.results = data.Results;
+                   $scope.winners = data.Winners;
+
+                   var resultsTrimSize = window.innerWidth / 30;
+                   var chartData = ResultsService.createDataTable(data.Results, resultsTrimSize);
+
+                   $scope.resultsBarChart = createBarChart(chartData, data.PollName);
+                   if (!$scope.loaded) {
+                       $scope.loaded = true;
```

Comment:
Minor point here, but I believe the check and update of `$scope.loaded` can be moved out of `updateResults()`, by changing line 35 into
```javascript
return ResultsService.getResults($scope.pollId)
```
This should then allow the call to `updateResults()` on line 24 to become
```javascript
updateResults()
.then(function(){$scope.loaded = true;});
```
which is a single update when we know it's appropriate, rather than checking every time,.